src/myr/Myr.js

```diff
@@ -670,7 +680,64 @@ class Myr {
     setTextureColoring = (i) => {
         this.cursor.textureColoring = Boolean(i);
         return this.cursor.textureColoring;
-    };
+    }
+
+    /**
+     * Changes the material type of the cursor.
+     * 
+     * @param {string} materialtype 
+     * @returns 
+     */
+    setMaterialType = (materialtype = MaterialType.SPECULAR) => {
+        let validMaterial = false;
+        for(let key in MaterialType) {
+            validMaterial = MaterialType[key] === materialtype;
+            if(validMaterial) break;
+        }
+        if(!validMaterial) {
+            throw new Error(`${materialtype} is not a valid material type!`);
+        }
+
+        this.cursor.materialtype = materialtype;
+        return this.cursor.materialtype;
+    }
+
+    /**
+     * Changes the roughness of the cursor within a range of 0 to 1.
+     * Custom roughness only applies to objects with MaterialType.PHYSICAL.
+     * 
+     * @param {number} roughness 
+     * @returns the roughness of the cursor
+     */
+    setRoughness = (roughness) => {
+        if(roughness < 0 || roughness > 1) {
+            throw new Error("Roughness must be between 0.0 and 1.0!");
+        }
+        this.cursor.roughness = roughness;
+        return this.cursor.roughness;
+    }
+
+    /**
+     * Changes the metalness of the cursor within a range of 0 to 1.
+     * Custom metalness only applies to objects with MaterialType.PHYSICAL.
+     * 
+     * @param {number} metalness 
+     * @returns the metalness of the cursor
+     */
+    setMetalness = (metalness) => {
+        if(metalness < 0 || metalness > 1) {
+            throw new Error("Metalness must be between 0.0 and 1.0!");
+        }
+        this.cursor.metalness = metalness;
+        return this.cursor.metalness;
+    }
+
+    /**
+     * @returns the list of material types
+     */
+    MaterialType = () => {
```

Comment:
is there a reason why this needs to be a function?

Reply:
That was for an object that essentially acted as an enum for the different material types that was added before I realized I could just add autocompletion for the strings. Of course with the autocompleter for the strings this isn't really necessary, and in hindsight would really only make sense in a more strongly typed language, as in JS nothing would stop you from trying to read from a nonexistent member anyway. I'll remove it so that inputs for setMaterial() are more consistent and streamlined.